src/main/kotlin/no/nav/klage/oppgave/clients/MicrosoftGraphClient.kt

```diff
@@ -27,7 +32,56 @@ class MicrosoftGraphClient(private val microsoftGraphWebClient: WebClient) {
             }.header("Authorization", "Bearer $accessToken")
 
             .retrieve()
-            .bodyToMono<MicrosoftGraphResponse>()
+            .bodyToMono<MicrosoftGraphIdentResponse>()
             .block()?.onPremisesSamAccountName ?: throw RuntimeException("NavIdent could not be fetched")
     }
+
+    fun getNamesForSaksbehandlere(identer: Set<String>, accessToken: String): Map<String, String> {
+        logger.debug("Fetching names for saksbehandlere from Microsoft Graph")
+
+        val identerNotInCache = identer.toMutableSet()
+        identerNotInCache -= saksbehandlerNameCache.keys
+        logger.debug("Only fetching identer not in cache: {}", identerNotInCache)
+
+        val chunkedList = identerNotInCache.chunked(MAX_AMOUNT_IDENTS_IN_GRAPH_QUERY)
```

Comment:
.chunked() kjente jeg ikke til, nå lærte jeg noe i dag også! :)

Reply:
Kotlin and extension functions ftw :)